<launch>

    <arg name="set_pose_topic" default="/virtual_master/set_pose"/>
    <arg name="cmd_vel_topic" default="/virtual_master/cmd_vel"/>
    <arg name="master_pose_topic" default="/virtual_master/master_pose"/>
    <arg name="master_vel_topic" default="/virtual_master/master_vel"/>

    <node name="virtual_master" type="virtual_master.py" pkg="formation_controller" output="screen">
            <rosparam command="load" file="$(find formation_controller)/config/virtual_master_config.yaml"/>
            <param name="set_pose_topic" value="$(arg set_pose_topic)" />
            <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)" />
            <param name="master_pose_topic" value="$(arg master_pose_topic)" />
            <param name="master_vel_topic" value="$(arg master_vel_topic)" />
    </node>


    <group ns="virtual_master">
    
        <arg name="robot_urdf_path" value="$(find mir_description)/urdf/mir_200/mir_200.gazebo.urdf"/>
        <arg name="model_name" default="virtual_master"/>
        <arg name="tf_prefix" default="virtual_master"/>
        <param name="robot_description" command="$(find xacro)/xacro $(arg robot_urdf_path) robot_namespace:=$(arg model_name) tf_prefix:=$(arg tf_prefix)" />

        

        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" >
            <param name="tf_prefix" value="virtual_master"/>
        </node>

        <node name="base_link_broadcaster" pkg="transport_launcher" type="base_broadcaster.py">
            <remap from="robot_pose" to="master_pose"/>
            <param name="tf_prefix" value="virtual_master"/>
            </node>  

            <node name="fake_mir_joint_publisher" pkg="mir_driver" type="fake_mir_joint_publisher.py" output="screen" />    
    </group>  

</launch>
